nf_core/utils.py

```diff
@@ -701,16 +701,28 @@ def get_tag_date(tag_date):
                 images = response.json()["images"]
                 singularity_image = None
                 docker_image = None
+                all_docker = {}
+                all_singularity = {}
                 for img in images:
-                    # Get most recent Docker and Singularity image
+                    # Get all Docker and Singularity images
                     if img["image_type"] == "Docker":
-                        modification_date = get_tag_date(img["updated"])
-                        if not docker_image or modification_date > get_tag_date(docker_image["updated"]):
-                            docker_image = img
-                    if img["image_type"] == "Singularity":
-                        modification_date = get_tag_date(img["updated"])
-                        if not singularity_image or modification_date > get_tag_date(singularity_image["updated"]):
-                            singularity_image = img
+                        # Obtain version and build
+                        match = re.search(r"(?::)+([A-Za-z\d\-_.]+)", img["image_name"])
+                        if match is not None:
+                            all_docker[match.group(1)] = {"date": get_tag_date(img["updated"]), "image": img}
+                    elif img["image_type"] == "Singularity":
+                        # Obtain version and build
+                        match = re.search(r"(?::)+([A-Za-z\d\-_.]+)", img["image_name"])
+                        if match is not None:
+                            all_singularity[match.group(1)] = {"date": get_tag_date(img["updated"]), "image": img}
+                # Obtain common builds from Docker and Singularity images
+                common_keys = list(all_docker.keys() & all_singularity.keys())
+                for k in common_keys:
+                    # Get the most recent common image
+                    if not docker_image or all_docker[k]["date"] > get_tag_date(docker_image["updated"]):
+                        docker_image = all_docker[k]["image"]
+                    if not singularity_image or all_singularity[k]["date"] > get_tag_date(singularity_image["updated"]):
+                        singularity_image = all_singularity[k]["image"]
```

Comment:
```suggestion
                current_date = datetime.datetime(0, 0, 0, 0, 0)
                for k in common_keys:
                    # Get the most recent common image
                    date = max(get_tag_date(all_docker[k]["date"]]), get_tag_date(all_docker[k]["date"]))
                    if docker_image is None or current_date < date:
                        docker_image = all_docker[k]["image"]
                        singularity_image = all_singularity[k]["image"]
                        current_date = date
```
This should guarantee that we get the same image for both Singularity and Docker.

Reply:
Thanks for the suggestion!